File: Projekt4/frontend/src/types/Fitness.ts
export interface FitnessClass {
  id: number;
  name: string;
  description: string;
  instructor: string;
  datetime: string;
  capacity: number;
  free_spots: number;
  created_at: string;
}

export interface Signup {
  id: number;
  class_id: number;
  user_id: number;
  is_waitlisted: boolean;
  signed_up_at: string;
}

export interface ClassCreate {
  name: string;
  description: string;
  instructor: string;
  datetime: string;
  capacity: number;
}

export interface ClassUpdate {
  name?: string;
  description?: string;
  instructor?: string;
  datetime?: string;
  capacity?: number;
}

export interface SignupCreate {
  class_id: number;
}

export interface MyClassSignup {
  id: number;
  class_id: number;
  user_id: number;
  is_waitlisted: boolean;
  signed_up_at: string;
  fitness_class: FitnessClass;
}

File: Projekt4/backend/app/schemas/fitness.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.schemas.user import UserOut


class ClassOut(BaseModel):
    id: int
    name: str
    description: str
    instructor: str
    datetime: datetime
    capacity: int
    created_at: datetime
    free_spots: int = 0

    model_config = {"from_attributes": True}


class ClassCreate(BaseModel):
    name: str
    description: str = ""
    instructor: str
    datetime: datetime
    capacity: int


class ClassUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    instructor: Optional[str] = None
    datetime: Optional[datetime] = None
    capacity: Optional[int] = None


class SignupOut(BaseModel):
    id: int
    class_id: int
    user_id: int
    is_waitlisted: bool
    signed_up_at: datetime

    model_config = {"from_attributes": True}


class SignupCreate(BaseModel):
    pass


class WaitlistEntry(BaseModel):
    signup_id: int
    user: UserOut
    signed_up_at: datetime

    model_config = {"from_attributes": True}


class UserClassOut(BaseModel):
    signup_id: int
    class_name: str
    instructor: str
    datetime: datetime
    is_waitlisted: bool
    signed_up_at: datetime
